patches/minecraft/net/minecraft/tileentity/TileEntityType.java.patch

```diff
@@ -18,3 +18,23 @@
           if (SharedConstants.field_206244_b) {
              throw illegalstateexception;
           }
+@@ -90,6 +90,10 @@
+       return (T)(this.field_200995_z.get());
+    }
+ 
++   public void addValidBlocks(Block ... blocks) {
++      this.field_223046_I.addAll(java.util.Arrays.asList(blocks));
```

Comment:
Not sure this is a good idea to open up, are you sure nothing relies on the constant nature of this list?

Reply:
Nothing relies on that list being constant.  I have to edit the furnace ones for FastFurnace to work properly, it's only used to check if the tile entity is actively on a valid block.

Reply:
`validBlocks` is a private variable and only its `contains` method is called in `isValidBlock`. So nothing relies on the constant nature.